# Data Cleaning for Video Games Sales Dataset


## Introduction 

Before we cleaned the dataset, it's better to get to know the dataset first. This dataset contains a list of video games with sales greater than 100,000 copies from 1980 to 2016. The dataset was taken from [Kaggle](https://www.kaggle.com/datasets/gregorut/videogamesales), where the author scrape the dataset from [vgchartz.com](https:///www.vgchartz.com).

Fields include

* `Rank` - Ranking of overall sales
* `Name` - The games name
* `Platform` - Platform of the games release (i.e. PC,PS4, etc.)
* `Year` - Year of the game's release
* `Genre` - Genre of the game
* `Publisher` - Publisher of the game
* `NA_Sales` - Sales in North America (in millions)
* `EU_Sales` - Sales in Europe (in millions)
* `JP_Sales` - Sales in Japan (in millions)
* `Other_Sales` - Sales in the rest of the world (in millions)
* `Global_Sales` - Total worldwide sales.


## Read the dataset

In [2]:
import pandas as pd


In [3]:
df = pd.read_csv("vgsales.csv")
df.head()


Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  object 
 2   Platform      16598 non-null  object 
 3   Year          16327 non-null  float64
 4   Genre         16598 non-null  object 
 5   Publisher     16540 non-null  object 
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.4+ MB


## Check for missing values

We can see from the dataset info above that there's some missing data on `Year` and `Publisher` column. Let's check out how many data is missing on each column.

In [5]:
df.isna().sum()


Rank              0
Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64

We got 271 missing data on `Year` column and 58 missing data on `Publisher` column. Since the amount of missing data is relatively small compared to the whole dataset (16598 rows), dropping the entire row that contain missing values won't affect our analysis by a significant margin. Let's drop the row using `.dropna()` method

In [6]:
df.dropna(subset=["Year", "Publisher"], axis=0, inplace=True)

## Check for Year range

The dataset should contain games that release from 1980 until 2016. So we have to make sure the `Year` column is in the range (1980 - 2016)

In [7]:
print(sorted(df["Year"].unique()))


[1980.0, 1981.0, 1982.0, 1983.0, 1984.0, 1985.0, 1986.0, 1987.0, 1988.0, 1989.0, 1990.0, 1991.0, 1992.0, 1993.0, 1994.0, 1995.0, 1996.0, 1997.0, 1998.0, 1999.0, 2000.0, 2001.0, 2002.0, 2003.0, 2004.0, 2005.0, 2006.0, 2007.0, 2008.0, 2009.0, 2010.0, 2011.0, 2012.0, 2013.0, 2014.0, 2015.0, 2016.0, 2017.0, 2020.0]


Year 2017 and 2020 shouldn't be in the dataset. Let's repeat the previous step to check how many rows that contain year 2017 and 2020 and remove them if it's only a small portion of the dataset. But first, convert the `Year` column from `float` into `int` to make the indexing easier.

In [8]:
df["Year"] = df["Year"].astype(int)
df.Year.value_counts()[[2017, 2020]]


2017    3
2020    1
Name: Year, dtype: int64

Year 2017 and 2020 only contain 4 rows combined so we can safely remove the row.

In [9]:
df = df[~df.Year.isin([2017, 2020])]


## Write the cleaned data to CSV file

Now that the dataset are cleaned, it's time to write the dataset back into CSV file and analyze it further.

In [10]:
df.to_csv("vgsales_cleaned.csv", index=False)